Jenkinsfile

```diff
@@ -239,6 +260,7 @@ pipeline {
                     }
                     post {
                         always {
+                            set_status(env.STAGE_NAME, currentResult)
```

Comment:
I think you are looking for `currentBuild.currentResult`.  Hopefully that one remains true on each stage and is not victim to JENKINS-57801.  If it is, we have that issue open with CouldBees to get fixed.  Not sure how long such a fix will take though.